File: frontend/src/services/category.ts
import api from './api'

export interface Category {
  id: number
  name: string
  parent_id: number | null
  description: string
  sort_order: number
  is_active: boolean
  children?: Category[]
}

export const categoryApi = {
  // 获取分类列表
  getCategories() {
    return api.get('/api/v1/categories')
  },

  // 获取分类详情
  getCategoryDetail(id: number) {
    return api.get(`/api/v1/categories/${id}`)
  },
}

File: backend/app/models/category.py
"""
分类模型
"""
from sqlalchemy import Column, Integer, String, Boolean, TIMESTAMP, Text, ForeignKey
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.core.database import Base


class Category(Base):
    """课程分类表"""
    __tablename__ = "categories"
    
    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    name = Column(String(100), nullable=False, comment="分类名称")
    parent_id = Column(Integer, ForeignKey("categories.id", ondelete="SET NULL"), nullable=True, index=True, comment="父分类ID")
    sort_order = Column(Integer, default=0, index=True, comment="排序")
    description = Column(Text, nullable=True, comment="分类描述")
    is_active = Column(Boolean, default=True, comment="是否激活")
    created_at = Column(TIMESTAMP, server_default=func.now(), comment="创建时间")
    updated_at = Column(TIMESTAMP, server_default=func.now(), onupdate=func.now(), comment="更新时间")
    
    # 关系
    parent = relationship("Category", remote_side=[id], backref="children")
    courses = relationship("Course", back_populates="category")
    
    def __repr__(self):
        return f"<Category(id={self.id}, name='{self.name}')>"




